lib/api_auth/railtie.rb

```diff
@@ -13,7 +13,40 @@ def api_authenticated?(secret_key)
         end
       end
 
+      module ClassMethods
+        def validation_with_api_auth(api_auth_options = nil)
+          ActionController.add_renderer(:json) do |json, options|
+            api_auth_options ||= options[:api_auth]
+            options.delete(:api_auth)
+
+            json = json.to_json(options) unless json.is_a?(String)
+
+            if options[:callback].present?
+              self.content_type = Mime[:js] if content_type.nil? || content_type == Mime[:json]
+
+              "/**/#{options[:callback]}(#{json})"
+            else
+              self.content_type ||= Mime[:json]
+
+              # API AUTH addition headers
+              if api_auth_options
+                response.headers['CONTENT-MD5'] ||= Digest::MD5.base64digest(json)
```

Comment:
Is this needed because the response object won't have a body for ApiAuth to inspect yet?